modules/media/php/media.class.inc

```diff
@@ -95,87 +86,60 @@ class Media extends \NDB_Menu_Filter
             $instrumentList[$instrument['Full_name']] = $instrument['Full_name'];
         }
 
-        $fileTypeList  = [];
+        $fileTypeList  = array();
         $fileTypeQuery = $db->pselect("SELECT file_type FROM media", []);
         foreach ($fileTypeQuery as $filetype) {
             $fileTypeList[$filetype['file_type']] = $filetype['file_type'];
         }
 
         //Language selector
         $languages    = \Utility::getLanguageList();
-        $languageList = [];
+        $languageList = array();
         foreach ($languages as $language) {
             $languageList[$language] = $language;
         }
 
-        $hiddenOptions = [];
-        if ($this->hasHidePermission) {
+        $hiddenOptions = array();
+        if (\User::singleton()->hasPermission('superuser')) {
             $hiddenOptions = array(
                               0 => 'Visible only',
                               1 => 'Hidden only',
                              );
         }
 
-        $this->fieldOptions = [
-                               'fileTypes'   => $fileTypeList,
-                               'visits'      => $visitList,
-                               'sites'       => $siteList,
-                               'instruments' => $instrumentList,
-                               'languages'   => $languageList,
-                               'hidden'      => $hiddenOptions,
-                              ];
-
-        return true;
+        return array(
```

Comment:
the documentation for this says it returns void

Reply:
Odd that it passes `phan` considering this... @HenriRabalais It may be a good idea to add a return type here and also to rebase to `major` if you haven't for a while